### DISCUSSION TITLE
Self host SDK with React Native

### DISCUSSION THREAD

**Discussion by ealeksandrov7** [Web3Auth/Web3Auth#571] [Help]
hHey guys. I am trying to integrate the self host SDK in a react native environment. Is that even possible? I guess it's built specifically for the web but are there any tricks/hacks I can use to succeed implementing it in a react native env? Using polyfills/shims, etc.. . Would really appreciate any input on this!

A bit more info on the "why":

What i need is to add a third share for the private key that must come from a password input. Sadly I don't think that's possible with the Plug and Play sdks or at least I haven't found a way to do it. Can I somehow change the "Seed Recovery 2fa" with a "Password 2fa" ?

**Comment by rinzler99** [comment]
Hey, 

You can use [this](https://web3auth.io/docs/sdk/self-host/modules/security-questions ) module to add "password 2fa" but only if you are using self-host sdk. For plug n play sdks this feature is in development and will be launched by next month.

The self host sdk supports only web right now.

Thanks!

**Reply by ealeksandrov7** [reply]
Actually I've managed to make the self host sdk work with my react native project and I used the module as you suggested for creating a password share. Though I'm looking forward to try out the password feature you're working on for the Plug n Play sdks.

Thanks for the answer!

**Reply by fess-v** [reply]
@rinzler99 did you integrate this module with plug n play sdks? can we already use it without making it self-hosted?
